src/ray/gcs/gcs_server/gcs_resource_manager.cc

```diff
@@ -262,6 +263,20 @@ void GcsResourceManager::HandleGetAllResourceUsage(
 
 void GcsResourceManager::UpdateNodeResourceUsage(const NodeID &node_id,
                                                  const rpc::ResourcesData &resources) {
+  if (auto maybe_node_info = gcs_node_manager_.GetAliveNode(node_id);
+      maybe_node_info != absl::nullopt) {
+    auto snapshot = maybe_node_info.value()->mutable_state_snapshot();
+    if (resources.idle_duration_ms() > 0) {
+      snapshot->set_state(rpc::NodeSnapshot::IDLE);
+      snapshot->set_idle_duration_ms(resources.idle_duration_ms());
+    } else {
+      if (resources.is_draining()) {
```

Comment:
i wonder if we should prioritize is_draining over the idle thing? 

like a draining node might still be reporting idle time i think?